src/Propel/Runtime/Connection/ConnectionWrapper.php

```diff
@@ -231,7 +231,7 @@ public function commit()
         $return = true;
         $opcount = $this->nestedTransactionCount;
 
-        if ($opcount > 0) {
+        if ($opcount > 0 && $this->inTransaction()) {
```

Comment:
It is. But in our case it's a bit about BC:
- in PHP 7.x there was a bug, that in case if you are inside transaction and execute any DDL query, you will still be inside transaction (that is not OK and controversial with docs), but most probably it just relates to a PDO internal "flag", while transaction is already auto-commited on DB level;
- in PHP 8 this bug was fixed, and since you call any DDL query, you will be auto-commited without any notices.
- in case we try to commit or rollback while not in transaction, it will lead to an exception.

Since we discovered such behavior only on PHP8, in this case, I suggest, we need to check whether we are `inTransaction` after every single query executed (ideally), and react somehow on it's change. In our case, it means we need to probably re-think all `$opcount` mechanism since we can't really rely on it anymore I assume. This approach is just to be BC with older versions of PHP.

@dereuromark is aware of this situation. From my perspective, we need a followup ticket here.

Reply:
Yeah, I just hope we don't introduce other issues with it.

Reply:
Created followup ticket.